phoenix-core/src/test/java/org/apache/phoenix/index/VerifySingleIndexRowTest.java

```diff
@@ -350,6 +351,7 @@ public void testVerifySingleIndexRow_allUnverified() throws IOException {
         }
     }
 
+    @Ignore
```

Comment:
Why are we ignoring tests rather than removing or fixing them?

Reply:
They will be fixed later. I will either fix it in this PR or let @swaroopak fix them in a separate jira. I needed to disable it to submit the patch for pre-commit testing

Reply:
@swaroopak and I worked together and fixed the failed unit tests. We needed to do some changes in the unit tests to adjust them for the changes of this PR. The unit tests caught couple issues in the code and we fixed them too. I now better appreciate this unit test :-)